src/crewai/agents/agent_adapters/base_tool_adapter.py

```diff
@@ -0,0 +1,32 @@
+from abc import ABC, abstractmethod
+from typing import Any, List, Optional
+
+from crewai.tools.base_tool import BaseTool
+
+
+class BaseToolAdapter(ABC):
+    """Base class for all tool adapters in CrewAI.
+
+    This abstract class defines the common interface that all tool adapters
+    must implement. It provides the structure for adapting CrewAI tools to
+    different frameworks and platforms.
+    """
+
+    original_tools: List[BaseTool] = []
+    converted_tools: List[Any] = []
```

Comment:
usually it's not recommend set mutable default argument like [], {} because they will be shared across  all instances of the class (i don't think you wanna that) 

so i'd recommend do it instead
```suggestion
def __init__(self):
        self.original_tools: List[BaseTool] = []
        self.converted_tools: List[Any] = []
```

Reply:
[here's an articles about that](https://docs.python-guide.org/writing/gotchas/)